File: js/268_MissingNumber.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
    var bit = nums.length;
    var i = 0;
    for(var j in nums){
        bit ^= nums[j];
        bit ^= i++;
    }
    return bit;
};

File: cpp/268_MissingNumber.cpp
//math
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n = nums.size() + 1;
        int sum = 0;
        for(auto num : nums){
            sum += num;
        }
        return (n-1)*n/2 - sum;
    }
};

//bit
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int bit = nums.size();
        int i = 0;
        for(auto num : nums){
            bit ^= num;
            bit ^= i;
            i++;
        }
        return bit;
    }
};